ebodac/src/main/java/org/motechproject/ebodac/service/impl/StatisticServiceImpl.java

```diff
@@ -44,79 +46,63 @@
 
     @Override
     public IvrAndSmsStatistic getStatisticForIvr(final Range<DateTime> dateRange) {
-        if (dateRange == null || dateRange.getMin() == null || dateRange.getMax() == null) {
-            return null;
-        }
+        return getStatisticForQuery(dateRange, IVR_STATISTIC_QUERY);
+    }
 
-        return ivrAndSmsStatisticReportDataService.executeSQLQuery(new SqlQueryExecution<IvrAndSmsStatistic>() {
+    @Override
+    public IvrAndSmsStatistic getStatisticForSms(final Range<DateTime> dateRange) {
+        return getStatisticForQuery(dateRange, SMS_STATISTIC_QUERY);
+    }
 
-            @Override
-            public IvrAndSmsStatistic execute(Query query) {
-                query.setResultClass(IvrAndSmsStatistic.class);
+    @Override
+    public List<IvrEngagementStatistic> getIvrEngagementStatistic() {
+        return ivrAndSmsStatisticReportDataService.executeSQLQuery(new SqlQueryExecution<List<IvrEngagementStatistic>>() {
 
-                Map<String, String> params = new HashMap<>();
-                params.put("minDate", dateRange.getMin().toString(DATE_TIME_FORMAT));
-                params.put("maxDate", dateRange.getMax().toString(DATE_TIME_FORMAT));
+            @Override
+            public List<IvrEngagementStatistic> execute(Query query) {
+                query.setResultClass(IvrEngagementStatistic.class);
 
-                List<IvrAndSmsStatistic> list = (List<IvrAndSmsStatistic>) query.executeWithMap(params);
-                if (list == null || list.isEmpty()) {
-                    return null;
-                }
-                return list.get(0);
+                return (List<IvrEngagementStatistic>) query.execute();
             }
 
             @Override
             public String getSqlQuery() {
-                return IVR_STATISTIC_QUERY;
+                return IVR_ENGAGEMENT_STATISTIC_QUERY;
             }
         });
     }
 
-    @Override
-    public IvrAndSmsStatistic getStatisticForSms(final Range<DateTime> dateRange) {
-        if (dateRange == null || dateRange.getMin() == null || dateRange.getMax() == null) {
-            return null;
-        }
-
-        return ivrAndSmsStatisticReportDataService.executeSQLQuery(new SqlQueryExecution<IvrAndSmsStatistic>() {
-
-            @Override
-            public IvrAndSmsStatistic execute(Query query) {
-                query.setResultClass(IvrAndSmsStatistic.class);
+    private IvrAndSmsStatistic getDataFromQuery(Query query, Range<DateTime> dateRange) {
```

Comment:
you are not getting any data from query, you can name this method "executeQueryWithParams" or something like that